test/e2e/apps/deployment.go

```diff
@@ -954,6 +970,112 @@ func testDeploymentsControllerRef(f *framework.Framework) {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+func testProportionalScalingDeployment(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	podLabels := map[string]string{"name": NginxImageName}
+	replicas := int32(10)
+
+	// Create a nginx deployment.
+	deploymentName := "nginx-deployment"
+	d := framework.NewDeployment(deploymentName, replicas, podLabels, NginxImageName, NginxImage, extensions.RollingUpdateDeploymentStrategyType)
+	d.Spec.Strategy.RollingUpdate = new(extensions.RollingUpdateDeployment)
+	d.Spec.Strategy.RollingUpdate.MaxSurge = intOrStrP(3)
+	d.Spec.Strategy.RollingUpdate.MaxUnavailable = intOrStrP(2)
+
+	framework.Logf("Creating deployment %q", deploymentName)
```

Comment:
For error messages, we start with lower case. For logging messages, will it be fine to start with capital case? Other tests have capital case for the logging messages.

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Error strings aren't capitalized because they're often prefixed before printing (you've probably seen `fmt.Errorf("xxx: %v", err)` a lot). This is not the case for logs. Logs should be capitalized.

Reply:
Thanks for the clarification! :)